src/audio/module_adapter/module_adapter_ipc4.c

```diff
@@ -48,7 +48,7 @@ int module_adapter_init_data(struct comp_dev *dev,
 		return -EINVAL;
 
 	dst->base_cfg = cfg->base_cfg;
-	dst->size = cfgsz;
+	dst->size = cfgsz - sizeof(cfg->base_cfg);
```

Comment:
@singalsu it's probably worth while having a generic accessor macro in the module API header so that IPC4 can be easily extracted

Reply:
I think I'm missing still cases when there's the extended config with pins appended (set in rimage for some components). I'm especially not happy with SRC component code change below. The cfg_size_expect gets value 4, from subtract 44 - 40. Also I'm at the moment testing a way to pass in testbench ipc4 build the UUID to avoid the uuid -> rimage id -> uuid translations with normal firmware. So there could be even more varying init ipc sizes.

Do you have ideas how to do this in a nicer way?